rfcs/0094-use-matrix-for-official-chat.md

```diff
@@ -0,0 +1,62 @@
+---
+feature: use-matrix-for-official-chat
+start-date: 2020-05-19
+author: Ryan Mulligan
+shepherd-team: @grahamc, @joepie91, @andir, @mweinelt
+shepherd-leader: @grahamc
+---
+
+# Summary
+[summary]: #summary
+
+Switch from IRC ([Freenode](https://freenode.net/)) to
+[Matrix](https://matrix.org/)
+([#community:nixos.org][1]) as
+our official community chat.
+
+# Motivation
+[motivation]: #motivation
+
+The current official community chat host, Freenode, has lost its
+leadership and its future is uncertain. Many Free Software communities
+are abandoning it, so its network effects may be negative instead of
+positive.
+
+There are currently 1796 members of the Unofficial NixOS Discord
+channel (244 weekly visitors, 95 weekly communicators). Some of them
+might be convinced by the more modern features available on Matrix to
+move their conversations there.
+
+# Detailed design
+[design]: #detailed-design
+
+Update the [NixOS.org community
+page](https://nixos.org/community/index.html) to point to our Matrix
+Space ([#community:nixos.org][1]) that has already been created which you
+can join by visting the [invite
+link][1].
```

Comment:
```suggestion
Update the [NixOS.org community
page](https://nixos.org/community/index.html) to point to our Matrix
Space ([#community:nixos.org][1]), [hosted](https://element.io/element-home) by Element, which you
can join by visting the [invite
link][1].
```

Reply:
I mean, the RFC is mentioning existing infrastructure, so I think it's good to be clear about its current setup, at the point the RFC is being discussed. This could change in the future without rendering the RFC obsolete, IMO. 

The "without any disruption" part has a few caveats, as mentioned in the edit below, so why not be clear?

I could change it to "currently hosted" if it helps.

Reply:
The problem is that this is *not* the current setup. The community space is not hosted by Element. Rooms exist across all servers that have participants in the room. If you want to mention that there is currently a "nixos.org" Matrix homeserver hosted by EMS, that should be in a different sentence or paragraph.

Reply:
The `#community:nixos.org` depends on it being available at that server which is hosted by Element. Of course we could use a more stable link like https://matrix.to/#/!MKvhXlSTLGJUXpYuWF:nixos.org?via=nixos.org&via=matrix.org&via=nerdsin.space which is less dependent on the state of the nixos.org server.

But still I think this is a insignificant detail. If nixos.org goes down for whatever reason we can just update the docs to point at the space via another homeserver (until we stand up a new nixos.org server).